.github/workflows/check-docs-pr-preview.yml

```diff
@@ -0,0 +1,62 @@
+# This workflow checks that PRs modifying website docs include a preview link in the PR description.
+# It enforces KubeStellar's documentation PR best practices.
+
+name: Check Docs PR Preview Link
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+
+jobs:
+  check-preview-link:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - name: Check for Preview Link in PR Description (bash)
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          REPO: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          set -e
+          BODY=$(gh pr view "$PR_NUMBER" --repo "$REPO" --json body -q '.body')
+          DOC_BRANCH_REGEX='^doc-'
+          GITHUB_PAGES_URL="https://$(echo "$REPO" | cut -d'/' -f1).github.io/$(echo "$REPO" | cut -d'/' -f2)/$BRANCH"
+          PREVIEW_LINE=$(echo "$BODY" | grep -iE '^\s*Preview.*https?://')
```

Comment:
This is good enough between friends, but is not good enough for the open internet. Consider a comment line like the following.

> Preview not at https://evil.deed/s but rather at https://myorg.github.io/kubestellar/doc-strangelove

Let's leave checking the URL to a later PR, and focus this PR on the basics.